src/Core/Settings.cpp

```diff
@@ -5887,6 +5887,7 @@ Experimental data deduplication for SELECT queries based on part UUIDs
     MAKE_OBSOLETE(M, Bool, use_mysql_types_in_show_columns, false) \
     MAKE_OBSOLETE(M, Bool, s3queue_allow_experimental_sharded_mode, false) \
     MAKE_OBSOLETE(M, LightweightMutationProjectionMode, lightweight_mutation_projection_mode, LightweightMutationProjectionMode::THROW) \
+    MAKE_OBSOLETE(M, Bool, skip_download_if_exceeds_query_cache, true) \
```

Comment:
Let's keep the old one as an alias instead, that way old queries/test will keep working but publicly and internally we'll use the clearer name